### PR TITLE
[Keyboard] Add mokey12x2

### PR DESCRIPTION
**Pull request by rhmokey**
## Description
[Keyboard] Add mokey12x2

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
> Also, you should not close and reopen PRs

You have previously been told not to do this. Is this an attempt to bump it to the top of the PR list?

**Comment by rhmokey** [comment]
> > Also, you should not close and reopen PRs
> 
> You have previously been told not to do this. Is this an attempt to bump it to the top of the PR list?

Sorry, I just want to know why the files can not be merged when there is no problem, and there is no administrator to follow up. Why?

**Comment by rhmokey** [comment]
> > Also, you should not close and reopen PRs
> 
> You have previously been told not to do this. Is this an attempt to bump it to the top of the PR list?

I observe that new files are merged every day, and none of them are new keyboards for contributors, they are ignored, including mine, is there any problem with qmk? When is it normal to merge new keyboards?

**Comment by fauxpark** [comment]
We review PRs when we have time, this is all volunteer run. Please just have some patience.

**Comment by zvecr** [comment]
> **When will my pull request be merged?**
You asked about the status of your Pull Request.
> 
> While we know that your PR is important to you, it may take some time for QMK collaborators to get to it.
> 
> PR reviews are something that is done in their free time -- collaborators are not paid nor compensated for the time spent reviewing, as it is a labor of love. As such, this means that it can take time for them to get to your PR -- things like family, or life can get in the way of finding time to perform reviews, and burnout is a serious consideration. The QMK firmware repository averages 200 PRs opened and 200 PRs merged every month, so please have patience.
> 
> Also, anyone can review PRs -- adding meaningful reviews to other PRs is a great way to contribute back. This helps reduce the burden on collaborators, meaning they may be able to get to your PR that much faster.

All you are doing by opening/closing additional PRs is creating extra noise, and making the project harder to maintain. Doing so will not help your PR get merged any faster.

**Comment by rhmokey** [comment]
> We review PRs when we have time, this is all volunteer run. Please just have some patience.

Thank you very much. I'll be patient

**Comment by rhmokey** [comment]
> > **When will my pull request be merged?**
> You asked about the status of your Pull Request.
> > 
> > While we know that your PR is important to you, it may take some time for QMK collaborators to get to it.
> > 
> > PR reviews are something that is done in their free time -- collaborators are not paid nor compensated for the time spent reviewing, as it is a labor of love. As such, this means that it can take time for them to get to your PR -- things like family, or life can get in the way of finding time to perform reviews, and burnout is a serious consideration. The QMK firmware repository averages 200 PRs opened and 200 PRs merged every month, so please have patience.
> > 
> > Also, anyone can review PRs -- adding meaningful reviews to other PRs is a great way to contribute back. This helps reduce the burden on collaborators, meaning they may be able to get to your PR that much faster.
> 
> All you are doing by opening/closing additional PRs is creating extra noise, and making the project harder to maintain. Doing so will not help your PR get merged any faster. If anything, I would personally actively ignore it.

Thank you very much. I'll be patient

File: keyboards/mokey/mokey12x2/halconf.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_SPI TRUE
```
Comment (drashna) [inline keyboards/mokey/mokey12x2/halconf.h:19]: Is there a reason that SPI is enabled? 

It doesn't appear to be used at all, so is unneeded.

File: keyboards/mokey/mokey12x2/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______
+            ),
+    [_FN2] = LAYOUT(
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______
+            ),
```
Comment (drashna) [inline keyboards/mokey/mokey12x2/keymaps/default/keymap.c:50]:
```suggestion
```

File: keyboards/mokey/mokey12x2/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______
+            ),
+    [_FN2] = LAYOUT(
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______
+            ),
```
Comment (drashna) [inline keyboards/mokey/mokey12x2/keymaps/via/keymap.c:50]:
```suggestion
```

File: keyboards/mokey/mokey12x2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+NKRO_ENABLE = no           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (drashna) [inline keyboards/mokey/mokey12x2/rules.mk:11]: These also could be move to the info.json
```suggestion
```

File: keyboards/mokey/mokey12x2/info.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "vid": "0x6653",
+        "pid": "0x012A",
+        "device_version": "0.0.1"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["A6", "A5", "A4", "A3", "B9"],
+        "rows": ["A2", "A1", "A0", "A7"]
+    },
+    "rgblight": {
```
Comment (drashna) [inline keyboards/mokey/mokey12x2/info.json:16]:
```suggestion
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgblight": true
    }
    "rgblight": {
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/mokey/mokey12x2/info.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{
+    "keyboard_name": "Mokey12x2",
+    "manufacturer": "Mokey",
+    "url": "https://github.com/rhmokey",
+    "maintainer": "Mokey",
+    "processor": "STM32F103",
+    "bootloader": "stm32-dfu",
```
Comment (fauxpark) [inline keyboards/mokey/mokey12x2/info.json:7]:
```suggestion
    "development_board": "bluepill",
```
If this is a Bluepill board, it should have the STM32Duino bootloader, not stm32-dfu, no?

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mokey/mokey12x2/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    [0] = LAYOUT(
+                KC_A,    KC_B,    KC_C,    KC_D,    KC_1,
+                KC_E,    KC_F,    KC_G,    KC_H,    KC_2,
+                KC_I,    KC_J,    KC_K,    KC_L,
+                KC_I,    KC_J,    KC_K,    KC_L),
+    [1] = LAYOUT(
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______),
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/keymaps/default/keymap.c:30]:
```suggestion
                KC_I,    KC_J,    KC_K,    KC_L)
```

File: keyboards/mokey/mokey12x2/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    [0] = LAYOUT(
+                KC_A,    KC_B,    KC_C,    KC_D,   KC_1,
+                KC_E,    KC_F,    KC_G,    KC_H,   KC_2,
+                KC_I,    KC_J,    KC_K,    KC_L,
+                KC_I,    KC_J,    KC_K,    KC_L),
+    [1] = LAYOUT(
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______,
+                _______, _______, _______, _______,
+                _______, _______, _______, _______),
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/keymaps/via/keymap.c:30]:
```suggestion
                KC_I,    KC_J,    KC_K,    KC_L)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mokey/mokey12x2/keymaps/via/keymap.c (hunk 8 above)
Comment (rhmokey) [reply keyboards/mokey/mokey12x2/keymaps/via/keymap.c]: > 

Problem solved

File: keyboards/mokey/mokey12x2/keymaps/default/keymap.c (hunk 7 above)
Comment (rhmokey) [reply keyboards/mokey/mokey12x2/keymaps/default/keymap.c]: > 
Problem solved

File: keyboards/mokey/mokey12x2/info.json (hunk 6 above)
Comment (rhmokey) [reply keyboards/mokey/mokey12x2/info.json]: > 如果这是 Bluepill 板，它应该有 STM32Duino 引导加载程序，而不是 stm32-dfu，不是吗？

Problem solved

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**